File: gachisa-backend/src/main/java/com/gachisa/order/repository/SavedDeliveryAddressRepository.java
package com.gachisa.order.repository;

import com.gachisa.order.entity.SavedDeliveryAddress;
import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface SavedDeliveryAddressRepository extends JpaRepository<SavedDeliveryAddress, Long> {
    List<SavedDeliveryAddress> findAllByBuyerIdOrderByUpdatedAtDesc(Long buyerId);
    Optional<SavedDeliveryAddress> findByIdAndBuyerId(Long id, Long buyerId);
}

File: gachisa-backend/src/main/kotlin/com/gachisa/order/repository/SavedDeliveryAddressRepository.kt
package com.gachisa.order.repository

import com.gachisa.order.entity.SavedDeliveryAddress
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface SavedDeliveryAddressRepository : JpaRepository<SavedDeliveryAddress, Long> {
    fun findAllByBuyerIdOrderByUpdatedAtDesc(buyerId: Long): List<SavedDeliveryAddress>
    fun findByIdAndBuyerId(id: Long, buyerId: Long): Optional<SavedDeliveryAddress>
}
